build/Dockerfile.apiserver

```diff
@@ -15,14 +15,13 @@
 FROM golang:1.22.2-bookworm as builder
 
 WORKDIR /workspace/src
-RUN git clone https://github.com/kubernetes/kubernetes --branch v1.23.2 --depth=1
+RUN git clone https://github.com/kubernetes/kubernetes --branch v1.30.1 --depth=1
```

Comment:
Any ideas why the env needs to rebuild it instead of using registry.k8s.io/kube-apiserver:v1.30.1 ?

Reply:
No, I don't see any reason for re-building . Your suggestion makes more sense to me than the current process. 
Is it OK, if I leave it like this in this PR, and you send a separate PR for your proposal?